### PR TITLE
Mark discriminator mapping as complete by default

### LINKED ISSUE #20192: Determine default for complete discriminator mapping
Till now, in terms of default, we always assumed that mapping is not complete (i.e. there can be rows in database with discriminator values we don't know). 
This issue tracks if we want to change that default.

### PR DESCRIPTION
**Pull request by smitpatel**
Resolves #20192

### CODE REVIEW

File: src/EFCore/Extensions/EntityTypeExtensions.cs [hunk 1]
```diff
@@ -658,7 +658,7 @@ public static bool GetIsDiscriminatorMappingComplete([NotNull] this IEntityType
                 return entityType.GetRootType().GetIsDiscriminatorMappingComplete();
             }
 
-            return (bool?)entityType[CoreAnnotationNames.DiscriminatorMappingComplete] ?? false;
+            return (bool?)entityType[CoreAnnotationNames.DiscriminatorMappingComplete] ?? true;
```
Comment (AndriySvyryd) [inline src/EFCore/Extensions/EntityTypeExtensions.cs]: Extract to `GetDefaultIsDiscriminatorMappingComplete`

Comment (smitpatel) [reply src/EFCore/Extensions/EntityTypeExtensions.cs]: private method?

Comment (AndriySvyryd) [reply src/EFCore/Extensions/EntityTypeExtensions.cs]: Yes

Comment (AndriySvyryd) [reply src/EFCore/Extensions/EntityTypeExtensions.cs]: This will make more sense when we have https://github.com/dotnet/efcore/issues/9329

**Pull request merged**